spikingjelly/activation_based/triton_kernel/flex_sn_inductor/hop.py

```
@@ -135,6 +221,560 @@ def eager_scan(
+    downstream compilers can decompose it to a loop instead of unrolling the
+    body T times in the traced graph. It is currently intended as an
+    experimental helper for investigation rather than a production default.
+
+    Notes:
+    * On the PyTorch versions we validate against, fake/proxy/export handling
+      for this out-of-tree scan pattern is still not stable enough to make this
+      the default compiled path.
+    * We pass flattened input/state leaves directly to ``torch.ops.higher_order.scan``
+      and rebuild the structured inputs inside ``wrapped_combine_fn``.
+    """
+    if _torch_scan_op is None or _wrap_scan_combine_fn_flat is None:
```

Ruff (0.15.11)

[error] Undefined name `_wrap_scan_combine_fn_flat`

(F821)